apps/server/src/schema/user/invite-operations.ts

```diff
@@ -214,23 +233,35 @@ builder.mutationFields((t) => ({
             frontEndInvitedUserUrl.search = searchParams.toString();
             logger.info(`Confirm invited non-existing user ${email}: ${frontEndInvitedUserUrl.toString()}`);
 
-            const newUser = await createInvitedUser(email, emailType, args.role, ctx.organizationId);
-            await Promise.all([
-              sendOrganizationInviteConfirmEmail({
-                firstName: newUser.firstName,
-                lastName: newUser.lastName,
-                email,
-                organizationName: organization.name,
-                expirationInDays: invitationExpirationInDays,
-                actionUrl: frontEndInvitedUserUrl.toString(),
-              }),
-              setConfirmInvitedUserRedis(inviteHash, newUser.id, ctx.organizationId),
-            ]);
+            const newUser: AError | User = await createInvitedUser(email, emailType, args.role, ctx.organizationId);
+            if (!(newUser instanceof AError)) {
+              await Promise.all([
+                sendOrganizationInviteConfirmEmail({
+                  firstName: newUser.firstName,
+                  lastName: newUser.lastName,
+                  email,
+                  organizationName: organization.name,
+                  expirationInDays: invitationExpirationInDays,
+                  actionUrl: frontEndInvitedUserUrl.toString(),
+                }),
+                setConfirmInvitedUserRedis(inviteHash, newUser.id, ctx.organizationId),
+              ]);
+            } else {
+              return new AError('Error creating invited user');
+            }
           }
         }),
       );
-      const flattenErrors = errors.flatMap((e) => e ?? []);
+
+      function isEmailMessageObject(
+        error: AError | { email: string; message: string },
+      ): error is { email: string; message: string } {
+        return !(error instanceof AError);
+      }
+      const flattenErrors = errors.flatMap((e) => e ?? []).filter(isEmailMessageObject);
```

Comment:
```suggestion
     const flattenErrors = errors.flatMap((e) => e ?? []);
```
you don't need these any more, errors return correct type

Reply:
![image](https://github.com/user-attachments/assets/3a7bce6e-ee89-47c2-8ba7-309a9b30b1b3)
Error: 
`Argument of type '(AError | { email: string; message: string; })[]' is not assignable to parameter of type '{ email: string; message: string; }[]'.
  Type 'AError | { email: string; message: string; }' is not assignable to type '{ email: string; message: string; }'.
    Property 'email' is missing in type 'AError' but required in type '{ email: string; message: string; }`
 
 Had to typeguard due to the above error.